src/qt/android/src/org/bitcoincore/qt/BitcoinQtActivity.java

```diff
@@ -25,5 +27,65 @@ public void onCreate(Bundle savedInstanceState)
         }
 
         super.onCreate(savedInstanceState);
+
+        // Marshmallow+ Permission APIs
+        if (Build.VERSION.SDK_INT >= 23) {
+             fuckMarshMallow();
+         }
+    }
+
+@Override
+public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+    switch (requestCode) {
+        case REQUEST_CODE_ASK_MULTIPLE_PERMISSIONS: {
+            Map<String, Integer> perms = new HashMap<String, Integer>();
+            perms.put(Manifest.permission.ACCESS_FINE_LOCATION, PackageManager.PERMISSION_GRANTED);
+            for (int i = 0; i < permissions.length; i++)
+                perms.put(permissions[i], grantResults[i]);
+            if (perms.get(Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+
+            ) {
+                 Toast.makeText(BitcoinQtActivity.this, "All Permission GRANTED!", Toast.LENGTH_SHORT).show();
+            } else {
+                 Toast.makeText(BitcoinQtActivity.this, "One or More Permissions are DENIED Exiting App!", Toast.LENGTH_SHORT).show();
+                 finish();
+            }
+        }
+        break;
+        default:
+            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
     }
 }
+
+@TargetApi(Build.VERSION_CODES.M)
+private void fuckMarshMallow() {
```

Comment:
I like swearing too but it belongs in the comments. Name the function after what it does.

Reply:
This was my mistake, I didn't pay attention to it, you are right.